### PR TITLE
Chore/test improvements

### PR DESCRIPTION
**Pull request by davidzwa**
Git gud... finally.

- [x] Run tests locally successfully (not the date offset, not a problem for now)
- [x] Run server successfully
- [x] Changelog

### CODE REVIEW

File: app-core.js [hunk 1]
```diff
@@ -65,42 +65,13 @@ async function ensureSystemSettingsInitiated() {
...
-    );
-  }
-
-  let listenerHttpServer = null;
-
-  logger.info("Initialising FarmInformation...");
-  await PrinterClean.initFarmInformation();
-
-  logger.info("Initialising Client Settings...");
   await ClientSettings.init();
```
Comment (davidzwa) [inline app-core.js:68]: These are now also initiated immediately.

File: app-core.js [hunk 2]
```diff
@@ -65,42 +65,13 @@ async function ensureSystemSettingsInitiated() {
...
-  const stateRunnerReport = await Runner.init();
-  logger.info("OctoFarm State returned", stateRunnerReport);
-
-  logger.info("Initialising SystemRunner...");
-  const sr = await SystemRunner.init();
-  logger.info("OctoFarm SystemRunner returned", sr);
-
-  await optionalInfluxDatabaseSetup();
-
-  listenerHttpServer = app.listen(port, "0.0.0.0", () => {
```
Comment (davidzwa) [inline app-core.js:97]: I moved the listen part back to `app.js` as this was the cause for headaches for API tests.

File: app-core.js [hunk 3]
```diff
@@ -151,14 +122,38 @@ async function serveOctoFarmNormally(app, port) {
...
+
+  return app;
 }
 
 const logger = new Logger("OctoFarm-Server");
 
 module.exports = {
   setupExpressServer,
   ensureSystemSettingsInitiated,
+  serveOctoFarmRoutes,
```
Comment (davidzwa) [inline app-core.js:157]: So we can serve the routes for tests. This is called by the serveOctoFarmNormally function, so both can be called.

File: app-fallbacks.js [hunk 4]
```diff
@@ -74,6 +74,17 @@ function serveNodeVersionFallback(app) {
   return listenerHttpServer;
 }
 
+function serveDatabaseIssueFallbackRoutes(app) {
```
Comment (davidzwa) [inline app-fallbacks.js:77]: Same idea for the database fallback: we dont have to boot up anything else for tests.

File: app.js [hunk 5]
```diff
@@ -13,44 +13,56 @@ try {
 
 if (!!majorVersion && majorVersion < 14) {
   const octoFarmServer = setupFallbackExpressServer();
-  return serveNodeVersionFallback(octoFarmServer);
-}
+  serveNodeVersionFallback(octoFarmServer);
+} else {
```
Comment (davidzwa) [inline app.js:17]: This else is the only change. No functional change.

File: app.js [hunk 6]
```diff
@@ -13,44 +13,56 @@ try {
...
+    .then(async () => {
+      const port = fetchOctoFarmPort();
+      if (!port || Number.isNaN(parseInt(port))) {
+        throw new Error(
+          "The server database-issue mode requires a numeric port input argument"
+        );
+      }
+
+      const app = await serveOctoFarmNormally(octoFarmServer);
+      app.listen(port, "0.0.0.0", () => {
```
Comment (davidzwa) [inline app.js:58]: Listening is done here for the normal serving. The fallbacks are still listening in the app-core file (node version, database connection issues)

File: test/api-views/dashboard.test.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+const { setupTestApp } = require("../../app-test");
+
+let request;
+
+beforeAll(async () => {
+  await dbHandler.connect();
+  const server = await setupTestApp();
+
+  request = supertest(server);
+});
```
Comment (davidzwa) [inline test/api-views/dashboard.test.js:12]: These lines are all we need to setup an API test now! Lot better if you ask me.

Comment (NotExpectedYet) [reply test/api-views/dashboard.test.js:12]: Yeah this is well easier, might even get me doing it more xD

**Review by NotExpectedYet — COMMENTED** [review]
Looks good man, lettuce move forward with this. I'll sort my settings one out in the morning will have conflicts.

**Review by NotExpectedYet — APPROVED** [review]

**Pull request merged**